modules/platform/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/BatchUploadFixture.java

```diff
@@ -240,42 +227,32 @@ private void itCanUseBatchUpload(boolean noDrop) throws IOException {
      */
     @Test
     public void testObeyFileTypeHeader() throws IOException {
-        Map<String, String> headers;
-
         // Get batch id, used as a session id
         String batchId = initializeDeprecatedNewBatch();
 
-        // Upload a file in multipart, first without the X-File-Type header, the second with
-        String fileName1 = "No header.txt";
+        // Upload a file without the X-File-Type header
         String data1 = "File without explicit file type";
-        String fileSize1 = String.valueOf(getUTF8Bytes(data1).length);
-        headers = new HashMap<>();
-        headers.put("X-File-Size", fileSize1);
-
-        try (FormDataMultiPart form = new FormDataMultiPart()) {
-            BodyPart fdp = new StreamDataBodyPart(fileName1, new ByteArrayInputStream(getUTF8Bytes(data1)));
-            form.bodyPart(fdp);
-            try (CloseableClientResponse response = getResponse(RequestType.POST, "upload/" + batchId + "/0", form,
-                    headers)) {
-                assertEquals(Status.CREATED.getStatusCode(), response.getStatus());
-            }
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "upload/" + batchId + "/0", data1,
+                Map.of("X-File-Name", "No header.txt"))) {
+            assertEquals(Status.CREATED.getStatusCode(), response.getStatus());
+            JsonNode node = mapper.readTree(response.getEntityInputStream());
+            assertEquals("true", node.get("uploaded").asText());
+            assertEquals(batchId, node.get("batchId").asText());
+            assertEquals("0", node.get("fileIdx").asText());
+            assertEquals("normal", node.get("uploadType").asText());
         }
 
-        String mimeType = "text/plain";
-        String fileName2 = "With header.txt";
+        // Upload a file with the X-File-Type header
         String data2 = "File with explicit X-File-Type header";
-        String fileSize2 = String.valueOf(getUTF8Bytes(data2).length);
-        headers = new HashMap<>();
-        headers.put("X-File-Size", fileSize2);
```

Comment:
Why not passing the file size header anymore?

Reply:
Because it's not checked or mandatory. 
If I didn't miss anything the only assertion made on the response is:
https://github.com/nuxeo/nuxeo/pull/4203/commits/cfca8399bb2c4f85a0f320a8dc8e15aed984fbf8#diff-a0e2882aaae16530623a620986daadc3L277